cmd/common/registry.go

```diff
@@ -213,17 +231,27 @@ func buildCommandFromGadget(
 
 				// This kind of gadgets return directly the result instead of
 				// using the parser
-				result, err := runtime.RunGadget(gadgetCtx)
+				results, err := runtime.RunGadget(gadgetCtx)
 				if err != nil {
 					return fmt.Errorf("running gadget: %w", err)
 				}
 
-				transformed, err := transformResult(result)
-				if err != nil {
-					return fmt.Errorf("transforming result: %w", err)
+				for node, result := range results {
+					transformed, err := transformResult(result)
+					if err != nil {
+						gadgetCtx.Logger().Warnf("transform result for %q: %v", node, result)
```

Comment:
>  IMO, in the CLI side, the logger should be mostly used for debugging, not to display errors

I think we can use the logger to show all kind of information, the only important thing is that those messages are visible to the user. Regarding the front end discussion here, probably the logger could be integrated with the frontend, so we're sure we send logs to the user?

Reply:
> I think we can use the logger to show all kind of information, the only important thing is that those messages are visible to the user. 
>
Isn't it strange that executing a command print a warning (only if the logger was configured), doesn't return an error, but doesn't print any result?

```bash
$ ./kubectl-gadget profile block-io
^CWARN[0003] transform result for "master": [123 34 118 ...]
$ echo $?
0
```

IMO, it is not clear what happened there 😅

Reply:
> only if the logger was configured

Are there cases when the logger is not configured?

I'd say the issue there is having a bad error message rather than how we're printing if, i.e. even if we print that message by returning an error it'll be confusing as well.

Reply:
> I'd say the issue there is having a bad error message rather than how we're printing if, i.e. even if we print that message by returning an error it'll be confusing as well.

Yeah, it was supposed to print `err`, not `result` (result might still be printed as debug, as that could help _us_).